message.go

```diff
@@ -951,6 +960,57 @@ func (m *RedisMessage) AsFtSearch() (total int64, docs []FtSearchDoc, err error)
 	panic(fmt.Sprintf("redis message type %c is not a FT.SEARCH response", typ))
 }
 
+type GeoLocation struct {
+	Name                      string
+	Longitude, Latitude, Dist float64
+	GeoHash                   int64
+}
+
+func (m *RedisMessage) AsGeosearch() ([]GeoLocation, error) {
+	arr, err := m.ToArray()
+	if err != nil {
+		return nil, err
+	}
+	geoLocations := make([]GeoLocation, 0, len(arr))
+	for _, v := range arr {
+		var loc GeoLocation
+		if v.IsString() {
+			loc.Name = v.string
+		} else {
+			info := v.values
+			var i int
+
+			//name
+			loc.Name = info[i].string
+			i++
+			//distance
+			if i < len(info) && info[i].IsFloat64() {
```

Comment:
We can’t use IsFloat64 here, since it only works with RESP3 and fails on RESP2. Checking the field not empty string string is enough.